scripts/bucketize.sh

```diff
@@ -0,0 +1,114 @@
+
+# This script takes the built Hugo site and:
+#   * creates a new S3 bucket named with either the current PR number or Git SHA
+#   * creates a list of all Hugo-generated client-side ("meta-refresh") redirects that
+#     we'll use to produce proper 301s later.
+#   * pushes the content of the website into the new S3 bucket.
+#   * tests the built website, first for broken links, then with Cypress to ensure pages
+#     render and behave properly.
+#   * emits a metadata file containing information about the commit and bucket, which
+#     Pulumi will use to process its update.
+#   * writes a record to AWS Parameter Store relating the generated bucket to the commit
+#     responsible for producing it.
+#   * Posts a PR comment back to GitHub, if applicable.
+
+source ./scripts/common.sh
+
+# The docroot of the built website.
+build_dir="public"
+
+# The text file we'll write as an output result.
+metadata_file="$(origin_bucket_metadata_filepath)"
+
+# Verify we have at least 1000 index.html files in total across the site.
+if [ ! "$(find $build_dir -type f | grep index.html | wc -l)" -ge 1000 ]; then
+    echo "Page-count check failed. Exiting."
+    exit 1
+fi
+
+# For previews, name the destination bucket with the PR number, to reduce the number of
+# buckets we create and to facilitate shorter sync times.
+destination_bucket="$(origin_bucket_prefix)-$(pr_number_or_git_sha)"
+destination_bucket_uri="s3://${destination_bucket}"
+
+# Translate Hugo redirects into a file we'll use for making 301 redirects later. Note that
+# we do this before syncing the site, as having the redirects file available for reference
+# can be helpful for debugging purposes.
+node scripts/translate-redirects.js "$build_dir" "$(pulumi -C infrastructure config get redirectDomain || echo '')"
+
+# Read the region from the stack's config -- we use it below.
+aws_region="$(pulumi -C infrastructure config get 'aws:region')"
+
+# Push site content to the bucket.
+echo "Synchronizing to $destination_bucket_uri..."
+
+# Make the bucket. If this fails, assume that means we've already made it.
+aws s3 mb $destination_bucket_uri --region $aws_region || echo "Bucket already exists. Continuing..."
+
+# Make the bucket an S3 website.
+aws s3 website $destination_bucket_uri --index-document index.html --error-document 404.html
+
+# Sync the local build directory to the bucket. Note that we do pass the --delete option
+# here, since in most cases, we'll be continually updating a bucket associated with a PR;
+# passing this option keeps the destination bucket clean.
+aws s3 sync "$build_dir" "$destination_bucket_uri" --acl public-read --delete --quiet
+
+echo "Sync complete."
+s3_website_url="http://${destination_bucket}.s3-website.${aws_region}.amazonaws.com"
+echo "$s3_website_url"
+
+# Set the content-type of latest-version explicitly. (Otherwise, it'll be set as binary/octet-stream.)
+aws s3 cp "$build_dir/latest-version" "${destination_bucket_uri}/latest-version" \
```

Comment:
`latest-version` is the file you referenced earlier in the comments that would be used by Pulumi?

Reply:
No, `latest-version` is the file used by the Pulumi CLI to determine the current released version of Pulumi.